File: 392-Is-Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i , j , m = t.size() ,  n  = s.size();
        for(i=0,j=0;i<m && j< n;i++){
            if(s[j] == t[i]){
                j++;
            }
        }
        if(j==n){
            return true ;
        }
        return false;
    }
};

File: 392-Is-Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int n = t.length()  , m = s.length();
        int i = 0  ;
        if(m==0 && n>0){
            return true;
        }
        if(m==0 && n==0){
            return true;
        }
        if(m>0 && n==0){
            return false;
        }
        char arrs[] = s.toCharArray() ;
        char arrt[] = t.toCharArray() ;
        for(int j=0;j<n;j++){
            if(arrt[j]==arrs[i]){
                i++;
                if(i==m){
                    return true;
                }
            }
        }
        return false ;
    }
}
